packages/rich-text/src/RichTextEditor.tsx

```diff
@@ -1,47 +1,120 @@
+import { withItalicEvents } from './plugins/Italic';
+import { withCodeEvents } from './plugins/Code';
+import { withMarksPlugin } from './plugins/Marks';
+import { withUnderlineEvents } from './plugins/Underline';
+import { Leaf } from './plugins/Leaf';
+
+// Import Contentful types.
+import { ContentfulEditor } from './types';
+
+// Import misc.
+import { compose } from './helpers/compose';
 
 type CustomElement = {
   type: 'paragraph';
-  children: CustomText[]
-}
-type CustomText = { text: string }
+  children: CustomText[];
+};
+type CustomText = { text: string };
 
 declare module 'slate' {
   interface CustomTypes {
-    Editor: BaseEditor & ReactEditor
-    Element: CustomElement
-    Text: CustomText
+    Editor: BaseEditor & ReactEditor;
+    Element: CustomElement;
+    Text: CustomText;
   }
 }
 
+const withPlugins = compose(withMarksPlugin, withReact);
+
+const withEvents = (editor) => (event) =>
+  [withBoldEvents, withItalicEvents, withCodeEvents, withUnderlineEvents].forEach((fn) =>
+    fn(editor, event)
+  );
+
 // Define our app...
 const RichTextEditor = () => {
-  const editor = useMemo(() => withReact(createEditor()), [])
+  const editor = useMemo<ContentfulEditor>(() => withPlugins(createEditor()), []);
 
   // Keep track of state for the value of the editor.
   const [value, setValue] = useState([
     {
       type: 'paragraph',
-      children: [{ text: 'A line of text in a paragraph.' }],
+      children: [
+        { text: 'This is editable ' },
+        { text: 'rich', bold: true },
+        { text: ' text, ' },
+        { text: 'much', italic: true },
+        { text: ' better than a ' },
+        { text: '<textarea>', code: true },
+        { text: '!' },
+      ],
+    },
+    {
+      type: 'paragraph',
+      children: [
+        {
+          text: "Since it's rich text, you can do things like turn a selection of text ",
+        },
+        { text: 'bold', bold: true },
+        {
+          text:
+            ', or add a semantically rendered block quote in the middle of the page, like this:',
+        },
+      ],
     },
-  ] as CustomElement[])
+    {
+      type: 'block-quote',
+      children: [{ text: 'A wise quote.' }],
+    },
+    {
+      type: 'paragraph',
+      children: [{ text: 'Try it out for yourself!' }],
+    },
+  ] as CustomElement[]);
+
+  const renderElement = useCallback((props) => {
+    switch (props.element.type) {
+      // TODO: add the components for `code`, `paragraph`, `image`, etc
+      default:
+        return <DefaultElement {...props} />;
+    }
+  }, []);
+
+  const renderLeaf = useCallback((props: RenderLeafProps) => {
+    return <Leaf {...props} />;
+  }, []);
 
   // Render the Slate context.
   return (
     <Slate
       editor={editor}
       value={value}
-      onChange={newValue => setValue(newValue as CustomElement[])}>
-      <Editable />
+      onChange={(newValue) => setValue(newValue as CustomElement[])}>
+      <Editable
```

Comment:
@m10l this is where we will add the toolbar. Example: https://github.com/ianstormtaylor/slate/blob/main/site/examples/richtext.tsx#L32

Reply:
Lovely, ty